### PR TITLE
Update to go-data-transfer v2 and libp2p, still wip

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Update to the new version of go-data-transfer to unblock other deps updates

# Status

- needs libp2p vet
- stuck on update go-index-provider

### CODE REVIEW

File: node/modules/lp2p/rcmgr.go [hunk 1]
```diff
@@ -111,10 +113,12 @@ func ResourceManager(connMgrHi uint) func(lc fx.Lifecycle, repo repo.LockedRepo)
 		if err != nil {
 			return nil, fmt.Errorf("error creating resource manager stats reporter: %w", err)
 		}
-		err = view.Register(obs.DefaultViews...)
-		if err != nil {
-			return nil, fmt.Errorf("error registering rcmgr metrics: %w", err)
-		}
+
+		// TODO: Libp2p converted to native prometheus metrics -- how do we register them
```
Comment (MarcoPolo) [inline node/modules/lp2p/rcmgr.go:117]: Looks like you're using the [default prometheus register](https://github.com/filecoin-project/lotus/blob/4588523de72452355fb6338176dbab6e69c5853a/cmd/lotus-stats/main.go#L159). You should be able to just do: https://github.com/libp2p/go-libp2p/blob/master/dashboards/resource-manager/README.md

e.g. `rcmgrObs.MustRegisterWith(prometheus.DefaultRegisterer)`

File: node/modules/lp2p/rcmgr.go [hunk 2]
```diff
@@ -52,37 +50,41 @@ func ResourceManager(connMgrHi uint) func(lc fx.Lifecycle, repo repo.LockedRepo)
 		// For every extra 1GB of memory we have available, increase our limit by 1GiB
 		defaultLimits.SystemLimitIncrease.Memory = 1 << 30
 		defaultLimitConfig := defaultLimits.AutoScale()
-		if defaultLimitConfig.System.Memory > 4<<30 {
+
+		changes := rcmgr.PartialLimitConfig{}
```
Comment (MarcoPolo) [inline node/modules/lp2p/rcmgr.go:54]: The diff in this file looks good to me for the rcmgr type changes. Sorry for the work, but hopefully it removes a foot gun.

File: chain/beacon/drand/drand_test.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func TestPrintGroupInfo(t *testing.T) {
 	})
 	chain, err := cg.FetchChainInfo(nil)
 	assert.NoError(t, err)
-	err = chain.ToJSON(os.Stdout)
+	err = chain.ToJSON(os.Stdout, nil)
```
Comment (marten-seemann) [inline chain/beacon/drand/drand_test.go:27]: Why use semver and then make breaking changes in a minor release? 🤬

File: node/modules/lp2p/rcmgr.go (hunk 1 above)
Comment (marten-seemann) [reply node/modules/lp2p/rcmgr.go:117]: Done.

Comment (marten-seemann) [reply node/modules/lp2p/rcmgr.go]: Actually, that doesn't work. It makes the tests panic, presumable when multiple hosts are started: https://app.circleci.com/pipelines/github/filecoin-project/lotus/27088/workflows/e4e8800d-6d83-4e2f-950f-3c619de1866a/jobs/874717/steps

Comment (MarcoPolo) [reply node/modules/lp2p/rcmgr.go]: ah right during a test you may spin up multiple of these.

Comment (marten-seemann) [reply node/modules/lp2p/rcmgr.go]: I added a `sync.Once`. Maybe not the nicest solution, but this should help us unblock this PR.

**Review by magik6k — APPROVED** [review]

**Pull request merged**